protocols/gossipsub/Cargo.toml

```diff
@@ -32,6 +32,7 @@ pin-project = "1.0.8"
 instant = "0.1.11"
 # Metrics dependencies
 open-metrics-client = "0.13.0"
+serde = { version = "1", optional = true }
```

Comment:
I think we should activate the `derive` feature here because we depend on it and at the moment, it seems to be carried along by something else but that might not always be true and can lead to compile errors.